### PR TITLE
Try to stabilize tests and update to AWS Provider 3.x

### PR DESCRIPTION
**Pull request by brikis98**
This is an attempt to stabilize the Vault tests. 

- Originally, they were failing due to intermittent issues downloading Terraform provider code; looks like the Terraform Registry is having some issues. Fixed with retries.
- Next, they were failing due to a lack of `t2.micro` in some regions. Fixed by excluding that region.

Now they are failing due to something on the server itself. [Example](https://circleci.com/gh/hashicorp/terraform-aws-vault/369?utm_campaign=build-failed&utm_medium=email&utm_source=notification). Typical output looks like this:

```
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: [0m[1mmodule.consul_cluster.aws_autoscaling_group.autoscaling_group: Creation complete after 48s [id=consul-test-7j3F6A2020041214375729150000000c][0m[0m
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: [0m[1m[32m
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: Apply complete! Resources: 48 added, 0 changed, 0 destroyed.[0m
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: [0m[1m[32m
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: Outputs:
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: 
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: asg_name_consul_cluster = consul-test-7j3F6A2020041214375729150000000c
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: asg_name_vault_cluster = vault-test-7j3F6A2020041214375542740000000b
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: auth_client_instance_id = i-0eba6c97a2f17d699
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: auth_client_public_ip = 54.209.106.53
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: auth_role_arn = arn:aws:iam::087285199408:role/vault-auth-test-7j3F6A-role20200412143742725500000002
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: aws_region = us-east-1
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: consul_cluster_cluster_tag_key = consul-test-7j3F6A
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: consul_cluster_cluster_tag_value = consul-test-7j3F6A
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: iam_role_arn_consul_cluster = arn:aws:iam::087285199408:role/consul-test-7j3F6A20200412143742720200000001
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: iam_role_arn_servers = arn:aws:iam::087285199408:role/consul-test-7j3F6A20200412143742720200000001
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: iam_role_arn_vault_cluster = arn:aws:iam::087285199408:role/vault-test-7j3F6A20200412143742740800000004
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: iam_role_id_consul_cluster = consul-test-7j3F6A20200412143742720200000001
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: iam_role_id_servers = consul-test-7j3F6A20200412143742720200000001
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: iam_role_id_vault_cluster = vault-test-7j3F6A20200412143742740800000004
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: launch_config_name_consul_cluster = consul-test-7j3F6A-20200412143745467600000009
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: launch_config_name_servers = consul-test-7j3F6A-20200412143745467600000009
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: launch_config_name_vault_cluster = vault-test-7j3F6A-2020041214375311420000000a
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: security_group_id_consul_cluster = sg-094b8e38431a4b2da
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: security_group_id_servers = sg-094b8e38431a4b2da
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: security_group_id_vault_cluster = sg-0cf9509938746e5b0
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: ssh_key_name = 7j3F6A
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: vault_cluster_size = 1
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: vault_servers_cluster_tag_key = Name
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: vault_servers_cluster_tag_value = vault-test-7j3F6A[0m
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z test_structure.go:23: The 'SKIP_validate' environment variable is not set, so executing stage 'validate'.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z save_test_data.go:195: Loading test data from /tmp/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami010344894/terraform-aws-vault/examples/vault-iam-auth/.test-data/TerraformOptions.json
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z retry.go:72: terraform [output -no-color auth_client_public_ip]
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:87: Running command terraform with args [output -no-color auth_client_public_ip]
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z command.go:158: 54.209.106.53
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:45Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:55Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:55Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:38:55Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:05Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:05Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:05Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:15Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:15Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:15Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:25Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:25Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:25Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:35Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:35Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:35Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:45Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:45Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:45Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:55Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:55Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:39:55Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:05Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:05Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:05Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:15Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:15Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:15Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:25Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:25Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:25Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:35Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:35Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:35Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:45Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:45Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:45Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:55Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:55Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:40:55Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:05Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:05Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:05Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:15Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:15Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:15Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:25Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:25Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:25Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:35Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:35Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:35Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:45Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:45Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:45Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:55Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:55Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:41:55Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:05Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:05Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:05Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:15Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:15Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:15Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:25Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:25Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:25Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:35Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:35Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:35Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:45Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:45Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:45Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:55Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:55Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:42:55Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:05Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:05Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:05Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:15Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:15Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:15Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:25Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:25Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:25Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:35Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:35Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:35Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:45Z retry.go:72: HTTP GET to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:45Z http_helper.go:27: Making an HTTP GET call to URL http://54.209.106.53:8080
TestMainVaultCluster/group/TestVaultIAMAuthWithvaultOpenSourceOnUbuntu16Ami 2020-04-12T14:43:45Z retry.go:84: HTTP GET to URL http://54.209.106.53:8080 returned an error: Get http://54.209.106.53:8080: dial tcp 54.209.106.53:8080: connect: connection refused. Sleeping for 10s and will try again.

[...]

http_helper.go:94: 'HTTP GET to URL http://54.209.106.53:8080' unsuccessful after 30 retries
```

Unfortunately, I'm now out of time. If anyone has a chance to pick this up, I'm guessing some script in User Data is failing, and we need the logs to see what's going on.

### CODE REVIEW

**Comment by brikis98** [comment]
Tests passed! Could I have a review plz?

File: modules/private-tls-cert/variables.tf [hunk 1]
```diff
@@ -74,20 +82,24 @@ variable "allowed_uses" {
...
 variable "private_key_ecdsa_curve" {
   description = "The name of the elliptic curve to use. Should only be used if var.private_key_algorithm is ECDSA. Must be one of P224, P256, P384 or P521."
   default     = "P256"
+  type        = string
 }
 
 variable "private_key_rsa_bits" {
   description = "The size of the generated RSA key in bits. Should only be used if var.private_key_algorithm is RSA."
   default     = "2048"
+  type        = string
```
Comment (yorinasub17) [inline modules/private-tls-cert/variables.tf:103]: NIT: we should use our consistent style of `description`, `type`, `default`. Probably do as another PR once we merge this.

Comment (brikis98) [reply modules/private-tls-cert/variables.tf:102]: May as well get the order right now: https://github.com/hashicorp/terraform-aws-vault/pull/195/commits/c031694781395dc29b2f8138685ebf48049a4f15.

**Comment by brikis98** [comment]
Thx for the review! Merging now.

**Comment by brikis98** [comment]
https://github.com/hashicorp/terraform-aws-vault/releases/tag/v0.13.9
